src/main/java/com/minecolonies/coremod/tileentities/TileEntityColonyBuilding.java

```diff
@@ -262,15 +309,59 @@ public NBTTagCompound writeToNBT(@NotNull final NBTTagCompound compound)
             //todo: actually do something about it and not spam the server
         }
         */
+        compound.setTag(TAG_INVENTORY, itemHandler.serializeNBT());
+        if (customName != null)
+        {
+            compound.setString(TAG_CUSTOM_NAME, customName);
+        }
         compound.setInteger(TAG_COLONY, colonyId);
         compound.setBoolean(TAG_MIRROR, mirror);
         return compound;
     }
 
-    @Override
     public boolean isUseableByPlayer(@NotNull final EntityPlayer player)
     {
-        return super.isUseableByPlayer(player) && this.hasAccessPermission(player);
+        return this.worldObj.getTileEntity(this.pos) != this
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Avoid if (x != y) ..; else ..; [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AConfusingTernary)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Ternary operators that can be simplified with || or && [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ASimplifiedTernary)